x-pack/plugins/infra/public/containers/logs/log_analysis/log_analysis_module.tsx

```diff
@@ -146,6 +146,27 @@ export const useLogAnalysisModule = <JobType extends string>({
     sourceId,
   ]);
 
+  useEffect(() => {
+    dispatchModuleStatus({
```

Comment:
I'm curious of how all of this connects together.

What does this effect do exactly to allow showing the tabs earlier?

Reply:
Previously, the module status reducer only used the source configuration it was initialized with to determine whether the job configuration is up-to-date. This meant that the source configuration had to be known before the provider was mounted.

Since this PR moves the loading state further down the hierarchy, the source configuration is now loaded in parallel to the job configuration. This tries to make sure that the reducer correctly evaluates the job configuration's currentness if the source configuration arrives asynchronously.

Does that make sense?